src/MultipartStreamBuilder.php

```diff
@@ -95,7 +95,14 @@ public function build()
                 $this->getHeaders($data['headers'])."\r\n";
 
             // Convert the stream to string
-            $streams .= (string) $data['contents'];
+            /* @var $contentStream StreamInterface */
+            $contentStream = $data['contents'];
+            if ($contentStream->isSeekable()) {
+                $streams .= (string) $data['contents'];
```

Comment:
Even though the output it the same. I think I would like to change this line to: 

```php
$streams .= $contentStream->_toString();
```
It will increase readability.